File: artisanat-backend/src/main/java/com/artisanat_backend/dto/ProductDto.java
package com.artisanat_backend.dto;

import com.artisanat_backend.enums.Category;
import com.artisanat_backend.enums.Collection;
import com.artisanat_backend.model.Artisan;
import lombok.Value;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * DTO for {@link com.artisanat_backend.model.Product}
 */
public class ProductDto {
    String name;
    String description;
    float price;
    int stock;
    LocalDateTime createdAt;
    Collection collection;
    Category category;
    Artisan artisan;
}

File: artisanat-frontend/src/app/core/dtos/product.dto.ts
import { ArtisanDTO } from './artisan.dto';
import { Collection } from '../enums/collection.enum';
import { Category } from '../enums/category.enum';

export interface ProductDto {
  name: string;
  description: string;
  price: number;
  stock: number;
  createdAt: string; // Use ISO string format
  collection: Collection;
  category: Category;
  artisan: ArtisanDTO;
}
